ffmpeg-plugin/mcm_video_tx.c

```diff
@@ -117,8 +133,17 @@ static int mcm_video_write_packet(AVFormatContext* avctx, AVPacket* pkt)
 static int mcm_video_write_trailer(AVFormatContext* avctx)
 {
     McmVideoMuxerContext *s = avctx->priv_data;
+    int err;
+
+    err = mesh_delete_connection(&s->conn);
+    if (err)
+        av_log(avctx, AV_LOG_ERROR, "Delete mesh connection failed: %s (%d)\n",
+               mesh_err2str(err), err);
+
+    err = mcm_put_client(&s->mc);
+    if (err)
+        av_log(avctx, AV_LOG_ERROR, "Put mesh client failed (%d)\n", err);
 
-    mcm_destroy_connection(s->tx_handle);
     return 0;
```

Comment:
Yeah, it doesn't make sense to return any error here since it's the `write_trailer()` function which is called once the media stream transmission ends. After that, FFmpeg exits. So, this return value doesn't affect the flow.

Reply:
ok. so maybe change of declaration to "void function_name()" ?

Reply:
Well, this function signature is predefined in FFmpeg. I can't change it :)
```
typedef struct FFOutputFormat {
...
    int (*write_header)(AVFormatContext *);
    /**
     * Write a packet. If FF_OFMT_FLAG_ALLOW_FLUSH is set in flags_internal,
     * pkt can be NULL in order to flush data buffered in the muxer.
     * When flushing, return 0 if there still is more data to flush,
     * or 1 if everything was flushed and there is no more buffered
     * data.
     */
    int (*write_packet)(AVFormatContext *, AVPacket *pkt);
    int (*write_trailer)(AVFormatContext *);
...
```